pyaerocom/colocateddata.py

```diff
@@ -350,13 +360,21 @@ def max(self):
         return self.data.max()
 
     def check_dimensions(self):
-        """Checks if data source and time dimension are at the right index"""
+        """Checks if data source and time dimension are at the right index
+
+        ToDo
+        ----
+        Check if this is needed. Little cumbersome at the moment, the data
+        object can / should be more flexible! Should
+        """
         dims = self.data.dims
         if not 2 < len(dims) < 5:
             logger.info('Invalid number of dimensions. Must be 3 or 4')
             return False
         try:
-            if dims.index('data_source') == 0 and dims.index('time') == 1:
+            if dims.index('data_source') == 0:
+                if 'time' in dims and dims.index('time') == 1:
+                    return True
                 return True
```

Comment:
This will return true even if there is no index called time.

```suggestion
            if dims.index('data_source') == 0:
                if 'time' in dims and dims.index('time') == 1:
                    return True
```

Reply:
Or is that the intended use? Then the check for the time dimension here is redundant.

```suggestion
            if dims.index('data_source') == 0:
                return True
```

Reply:
I think this whole method should go out, I checked locally and it is not really needed I think. I'll remove it and push the update.